deltachat-ios/Controller/WebxdcViewController.swift

```diff
@@ -271,7 +276,7 @@ class WebxdcViewController: WebViewViewController {
                let maxSerial = first["max_serial"] as? Int {
                 self.lastSerial = maxSerial
             }
-            webView.evaluateJavaScript("window.__webxdcUpdate(atob(\"\(statusUpdates.toBase64())\"))", completionHandler: nil)
+            webView.evaluateJavaScript("window.__webxdcUpdate(\(String(lastSerial)))", completionHandler: nil)
```

Comment:
at a first glance, i fear that here may be a race condition leading to double updates:

if after this line, core receives a new update, this will be returned by `webxdc-update.json?SERIAL` _without_ updating `lastSerial`. when then `updateWebxdc()` is called again, the same update is returned again.

but please @Simon-Laux , @cyBerta  double-check, might be that the race is impossible by some reasons i do not have in mind.

Reply:
ok let's move the whole last serial shenanigans to the js side to not have this issue?

Reply:
moved the stuf to js https://github.com/deltachat/deltachat-ios/pull/1718